File: week-5/6-the-full-counting-sort/main.swift
import Foundation

/*
 * Complete the 'countSort' function below.
 *
 * The function accepts 2D_STRING_ARRAY arr as parameter.
 */

func countSort(arr: [[String]]) -> Void {
    // Write your code here

}

guard let n = Int((readLine()?.trimmingCharacters(in: .whitespacesAndNewlines))!)
else { fatalError("Bad input") }

var arr = [[String]]()

for _ in 1...n {
    guard let arrRowTemp = readLine()?.replacingOccurrences(of: "\\s+$", with: "", options: .regularExpression) else { fatalError("Bad input") }

    let arrRow: [String] = arrRowTemp.split(separator: " ").map {
        String($0)
    }

    guard arrRow.count == 2 else { fatalError("Bad input") }

    arr.append(arrRow)
}

guard arr.count == n else { fatalError("Bad input") }

countSort(arr: arr)

File: week-5/6-the-full-counting-sort/main.kt
import java.io.*
import java.math.*
import java.security.*
import java.text.*
import java.util.*
import java.util.concurrent.*
import java.util.function.*
import java.util.regex.*
import java.util.stream.*
import kotlin.collections.*
import kotlin.comparisons.*
import kotlin.io.*
import kotlin.jvm.*
import kotlin.jvm.functions.*
import kotlin.jvm.internal.*
import kotlin.ranges.*
import kotlin.sequences.*
import kotlin.text.*

/*
 * Complete the 'countSort' function below.
 *
 * The function accepts 2D_STRING_ARRAY arr as parameter.
 */

fun countSort(arr: Array<Array<String>>): Unit {
    // Write your code here

}

fun main(args: Array<String>) {
    val n = readLine()!!.trim().toInt()

    val arr = Array<Array<String>>(n, { Array<String>(2, { "" }) })

    for (i in 0 until n) {
        arr[i] = readLine()!!.trimEnd().split(" ").toTypedArray()
    }

    countSort(arr)
}
